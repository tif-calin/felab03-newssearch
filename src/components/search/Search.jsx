import React from 'react';
import PropTypes from 'prop-types';

const Search = props => {
  return <div className="Search">
    <input 
      type="text" 
      defaultValue={props.query}
      onChange={e => props.setQuery(e.target.value)}
    />
  </div>;
};

Search.propTypes = {
  query: PropTypes.string.isRequired,
  setQuery: PropTypes.func.isRequired
};

export default Search;
