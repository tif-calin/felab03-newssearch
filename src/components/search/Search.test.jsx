import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());

  it('renders Search', () => {
    const { asFragment } = render(<Search query="" setQuery={() => true}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
