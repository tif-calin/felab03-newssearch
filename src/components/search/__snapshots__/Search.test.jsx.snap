// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <input
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
